source/adios2/operator/compress/CompressPNG.cpp

```diff
@@ -0,0 +1,220 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ *
+ * CompressPNG.cpp
+ *
+ *  Created on: Jun 10, 2019
+ *      Author: William F Godoy godoywf@ornl.gov
+ */
+#include "CompressPNG.h"
+
+#include <cstring> // std::memset
+
+extern "C" {
+#include <png.h>
+}
+
+#include "adios2/helper/adiosFunctions.h"
+
+namespace adios2
+{
+namespace core
+{
+namespace compress
+{
+
+const std::map<std::string, uint32_t> CompressPNG::m_ColorTypes = {
+    {"PNG_COLOR_TYPE_GRAY", PNG_COLOR_TYPE_GRAY},
+    {"PNG_COLOR_TYPE_PALETTE", PNG_COLOR_TYPE_PALETTE},
+    {"PNG_COLOR_TYPE_RGB", PNG_COLOR_TYPE_RGB},
+    {"PNG_COLOR_TYPE_RGB_ALPHA", PNG_COLOR_TYPE_RGB_ALPHA},
+    {"PNG_COLOR_TYPE_GRAY_ALPHA", PNG_COLOR_TYPE_GRAY_ALPHA},
+    {"PNG_COLOR_TYPE_RGBA", PNG_COLOR_TYPE_RGBA},
+    {"PNG_COLOR_TYPE_GA", PNG_COLOR_TYPE_GA}};
+
+const std::map<std::string, std::set<uint32_t>> CompressPNG::m_BitDepths = {
+    {"PNG_COLOR_TYPE_GRAY", {1, 2, 4, 8, 16}},
+    {"PNG_COLOR_TYPE_PALETTE", {1, 2, 4, 8}},
+    {"PNG_COLOR_TYPE_RGB", {8, 16}},
+    {"PNG_COLOR_TYPE_RGB_ALPHA", {8, 16}},
+    {"PNG_COLOR_TYPE_GRAY_ALPHA", {8, 16}},
+    {"PNG_COLOR_TYPE_RGBA", {8, 16}},
+    {"PNG_COLOR_TYPE_GA", {8, 16}}};
```

Comment:
It looks like you are doing a very full features exposure of the feature set of `libpng`, but from what I have read, basically all hardware is doing 8 bit bit depths. I haven't yet understood your intended use case, but are the variable bit lengths exposed to the user?